src/main/java/com/google/api/codegen/discovery/transformer/SampleTypeNameConverter.java

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.discovery.transformer;
+
+import com.google.api.codegen.discovery.SampleConfig;
+import com.google.api.codegen.discovery.TypeInfo;
+import com.google.api.codegen.util.TypeName;
+import com.google.api.codegen.util.TypedValue;
+
+/**
+ * Maps Type instances to TypeName instances.
+ */
+public interface SampleTypeNameConverter {
+
+  /**
+   * Provides a TypeName for the service.
+   */
+  TypeName getServiceTypeName(SampleConfig sampleConfig);
```

Comment:
This seems slightly out of place. It's not converting one type to another. My first instinct is to suggest putting it in SampleNamer.